xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -2886,6 +2857,34 @@ void CVideoPlayer::HandleMessages()
       auto pValue = std::static_pointer_cast<CDVDMsgBool>(pMsg);
       SetSubtitleVisibleInternal(pValue->m_value);
     }
+    else if (pMsg->IsType(CDVDMsg::SUBTITLE_A53_DATA))
+    {
+      auto sideData = std::static_pointer_cast<CDVDMsgSubtitleCCData>(pMsg);
+      if (m_pCCDemuxer)
+      {
+        while (!m_bAbortRequest)
+        {
+          DemuxPacket* pkt = m_pCCDemuxer->Process(sideData->m_ccData.release());
+          if (!pkt)
+            break;
+
+          if (m_pCCDemuxer->GetNrOfStreams() !=
+              m_SelectionStreams.CountTypeOfSource(STREAM_SUBTITLE, STREAM_SOURCE_VIDEOMUX))
+          {
+            m_SelectionStreams.Clear(STREAM_SUBTITLE, STREAM_SOURCE_VIDEOMUX);
+            m_SelectionStreams.Update(NULL, m_pCCDemuxer.get(), "");
+            UpdateContent();
+            OpenDefaultStreams(false);
+          }
+          CDemuxStream* pSubStream = m_pCCDemuxer->GetStream(pkt->iStreamId);
+          if (pSubStream && m_CurrentSubtitle.id == pkt->iStreamId &&
+              m_CurrentSubtitle.source == STREAM_SOURCE_VIDEOMUX)
+            ProcessSubData(pSubStream, pkt);
+          else
+            CDVDDemuxUtils::FreeDemuxPacket(pkt);
+        }
```

Comment:
Just wondering if theres going to be a leak from a "missing" FreeDemuxPacket. I tried to follow the ProcessSubData into the SendMessage, which (i think) goes into the VideoPlayer queue and i think is handled in https://github.com/xbmc/xbmc/blob/d3474a88f4a161101569f1c98bd866c17369fd6d/xbmc/cores/VideoPlayer/VideoPlayerVideo.cpp#L302

Through all that, i can see in the process that it looks like it would hit https://github.com/xbmc/xbmc/blob/d3474a88f4a161101569f1c98bd866c17369fd6d/xbmc/cores/VideoPlayer/VideoPlayerVideo.cpp#L528-L531, but theres no Free anywhere along the way.

Keep in mind, i really dont know the VP side of things, more of a query than anything.

Reply:
nm, the destructor of CDVDMsgDemuxerPacket has a free. all good :D

Reply:
Yeah the code in this regard is exactly the same as before, I just nuked the Read() operation since that data is already provided by ffmpeg in the side data, so we just need to feed the decoder with that data.

Makes me question if the parse closed captions setting makes any sense after this